extensions/gcp/controllerservices/GCPCredentialsControllerService.cpp

```diff
@@ -69,20 +70,24 @@ std::shared_ptr<gcs::oauth2::Credentials> GCPCredentialsControllerService::creat
 }
 
 void GCPCredentialsControllerService::onEnable() {
-  CredentialsLocation credentials_location;
-  if (!getProperty(CredentialsLoc, credentials_location)) {
-    logger_->log_error("Invalid Credentials Location, defaulting to %s", toString(CredentialsLocation::USE_DEFAULT_CREDENTIALS));
+  std::string value;
+  std::optional<CredentialsLocation> credentials_location;
+  if (getProperty(CredentialsLoc, value)) {
+    credentials_location = magic_enum::enum_cast<CredentialsLocation>(value);
+  }
+  if (!credentials_location) {
+    logger_->log_error("Invalid Credentials Location, defaulting to %s", magic_enum::enum_name(CredentialsLocation::USE_DEFAULT_CREDENTIALS).data());
```

Comment:
This should be safe as it points to a null terminated string according to [this](https://github.com/Neargye/magic_enum/issues/37#issuecomment-626593807)